react-carousel-component/src/index.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+const slides = [
+  {
+    src: 'https://cdn.betterttv.net/emote/5590b223b344e2c42a9e28e3/3x'
+  },
+  {
+    src: 'https://cdn.betterttv.net/emote/58ae8407ff7b7276f8e594f2/3x'
+  },
+  {
+    src: 'https://cdn.betterttv.net/emote/5aa1d0e311237146531078b0/3x'
+  },
+  {
+    src: 'https://cdn.betterttv.net/emote/5e4e7a1f08b4447d56a92967/3x'
+  },
+  {
+    src: 'https://cdn.betterttv.net/emote/5b77ac3af7bddc567b1d5fb2/3x'
+  }
+];
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { source: slides[0].src };
+    this.index = 0;
+    this.timedOut = false;
+    this.handleClick = this.handleClick.bind(this);
+    this.autoPan = setInterval(() => {
+      if (this.timedOut) {
+        return;
+      }
+      this.next();
```

Comment:
Use `componentDidMount` to kick things off. The constructor runs before the component has successfully been rendered for the first time and you don't want to have a memory leak in case that fails.